### PR TITLE
Add scripting capabilities and support for extensions

### PR DESCRIPTION
**Pull request by nexus6-haiku**

### CODE REVIEW

File: src/GenioApp.cpp [hunk 1]
```diff
@@ -461,6 +476,301 @@ GenioApp::_PrepareConfig(ConfigManager& cfg)
 }
 
 
+// Genio Scripting
```
Comment (Freaxed) [inline src/GenioApp.cpp:479]: Can we evaluate to move scripting related objects to another file?

File: src/ui/Editor.cpp [hunk 2]
```diff
@@ -1395,7 +1396,65 @@ Editor::GetSymbol()
...
+	SendMessage(SCI_ENDUNDOACTION, 0, 0);
+}
+
+
+const BString
+Editor::GetLine(int32 lineNumber)
+{
+	int32 lineLength = SendMessage(SCI_LINELENGTH, lineNumber, UNSET);
+	char *lineBuffer = new char[lineLength + 1];
+	memset(lineBuffer, '\0', lineLength + 1);
```
Comment (Freaxed) [inline src/ui/Editor.cpp:1442]: lineBuffer[lineLength]='\0' should be enough

File: src/ui/Editor.cpp [hunk 3]
```diff
@@ -1871,9 +1927,10 @@ Editor::_CommentLine(int32 position)
 	const std::string lineCommenter = fCommenter + ' ';
 	int32 lineNumber = SendMessage(SCI_LINEFROMPOSITION, position, UNSET);
 	int32 lineLength = SendMessage(SCI_LINELENGTH, lineNumber, UNSET);
-	char *lineBuffer = new char[lineLength];
+	char *lineBuffer = new char[lineLength + 1];
+	memset(lineBuffer, '\0', lineLength + 1);
 	SendMessage(SCI_GETLINE, lineNumber, (sptr_t)lineBuffer);
-	std::string line(lineBuffer);
+	std::string line(lineBuffer, lineLength);
```
Comment (Freaxed) [inline src/ui/Editor.cpp:1933]: Can we reuse the GetLine here?

File: src/GenioApp.cpp (hunk 1 above)
Comment (nexus6-haiku) [reply src/GenioApp.cpp:479]: I tried but reverted it back as I got too many errors. I plan to try again, though.

File: src/ui/Editor.cpp (hunk 2 above)
Comment (nexus6-haiku) [reply src/ui/Editor.cpp]: done

File: src/ui/Editor.cpp (hunk 3 above)
Comment (nexus6-haiku) [reply src/ui/Editor.cpp]: done

**Pull request merged**